src/bronto_burt.c

```diff
@@ -0,0 +1,900 @@
+#include "bronto_burt.h"
+#include "functions.h"
+#include "object.h"
+#include "task.h"
+
+struct Object2* CreateBrontoBurt(struct Object* arg0, u8 arg1) {
+    struct Object2 *obj, *obj2;
+    struct Task *task = TaskCreate(ObjectMain, 0xb4, 0x1000, 0x10, ObjectDestroy);
+    obj = TASK_GET_STRUCT_PTR(task, obj2);
+    InitObject(obj, arg0, arg1);
+    if(obj->x > obj->kirby3->base.x) {
+        obj->flags |= 1;
+    } else {
+        obj->flags &= ~1;
+    }
+    obj->unkC |= 1;
+    obj->unkC |= 4;
+    sub_0803E2B0(obj, -5, -6, 5, 5);
+    sub_0803E308(obj, -6, -7, 6, 7);
+    ObjectInitSprite(obj);
+    switch(arg0->subtype1) {
+        default:
+        case 0:
+            if(obj->y > obj->kirby3->base.y) {
+                sub_080AEB68(obj); // Done
+            } else {
+                sub_080AD8F0(obj); // Done
+            }
+            break;
+        case 1:
+            if(obj->y > obj->kirby3->base.y) {
+                sub_080AEBD8(obj); // Done
+            } else {
+                sub_080AEC34(obj); // Done
+            }
+            break;
+        case 2:
+            sub_080ADCA4(obj); // Done
+            break;
+        case 3:
+            sub_080AE100(obj); // Done 
+            break;
+        case 4:
+            sub_080AECA4(obj); // Done
+            break;
+        case 5:
+            sub_080AED1C(obj); // Done
+    }
+    obj->unk9E = 0;
+    obj->unk7C = 0;
+    return obj;
+}
+
+void sub_080AD7C0(struct Object2* obj) {
+    obj->flags = (obj->flags | 0x140) & ~0x20;
+    obj->yspeed = 0;
+    switch(obj->object->subtype1) {
+        default:
+        case 0:
+            if (obj->y > obj->kirby3->base.y) {
+                sub_080AEB68(obj);
+            } else {
+                sub_080AD8F0(obj);
+            }
+            break;
+        case 1:
+            if (obj->y > obj->kirby3->base.y) {
+                sub_080AEBD8(obj);
+            } else {
+                sub_080AEC34(obj);
+            }
+            break;
+        case 2:
+            if(obj->subtype != 0) {
+                sub_080AEC34(obj);
+            } else {
+                sub_080AD8F0(obj);
+            }
+            break;
+        case 3:
+            sub_080AE100(obj);
+            break;
+        case 4:
+            sub_080AECA4(obj);
+            break;
+        case 5:
+            sub_080AED90(obj);
+    }
+}
+
+void sub_080AD870(struct Object2* obj) {
+    if(obj->xspeed < 0) {
+        obj->xspeed += 0xE;
+        if(obj->xspeed > 0) {
+            obj->xspeed = 0;
+        }
+    } else {
+        obj->xspeed -= 0xE;
+        if(obj->xspeed < 0) {
+            obj->xspeed = 0;
+        }
+    }
+    obj->counter++;
+}
+
+void sub_080AD8AC(struct Object2* obj) {
+    obj->flags |= 4;
+    if((obj->yspeed -= 0x19) << 0x10 < -0x28A0000) {
```

Comment:
for this fakematching -- add a comment showing a clearer equivalent condition. same below.